### ISSUE TITLE
Level difference gives HUGE chance to miss with a spell in pvp

### ISSUE THREAD

**Issue by Crusaderz** [LightsHope/server#626]
First of all i'm sorry for my poor english.

The problem:
Chance to miss with a spell in pvp same as in pve. It shouldn't be like this. **Completely unplayable**.

Fighting against "brown" level player is rather uncomfortable since they resists spells often. "Red" level enemy resists close to 80% of spells casted on them. "Skull" - forget it.

Physical damage dealers don't have such problem even against "skull" level enemy.

Back in 2006 casters never had this issue. Meeting someone 5 levels higher doesn't mean you are definitely going to graveyard. Casters could kill red and **EVEN** skull enemy in world pvp. They were playing battlegrounds as level x0 and fight against +9 enemy who only had better stats and spells and couldn't dodge spells like the Neo. I had a warlock back in the days. I played BGs and never had spell miss issues at all.
Sure there was chance to miss based on a level difference and level 1 would miss with most of his casts vs 60 but against someone ~20 level higher spells would land most of the time.

Proofs:

45 hunter vs 54-60 rogues.
9-15 level difference. Rogues eat imp. concussive shot (the stun proc is a spell) and freezing traps. https://www.youtube.com/watch?v=GUXvC9wKUrk

36 or 39 hunter vs 60 rogue.
Atleast 21 level difference. Rogue resisted one imp. concussive shot out of 2 procs and ate freezing trap.
https://youtu.be/NqSd1tpt06o

48 priest vs 60.
Priest killing various level 60 players.
https://www.youtube.com/watch?v=iOjJY29HA1I
Video has hundreds commentaries, the oldest from 2006 and latest published couple months ago (and only him asked about resists):

> Uh, how are skull levels not resisting like 99% of your spells

clearly that poor guy is a private server victim.
Since the video quality is 240p i reuploaded it with better quality: https://youtu.be/InXWMGxOfEo

49 mage vs 60.
Various pvp footage:
https://youtu.be/qZD4KmR7tb8

49 warlock vs 60.
Various pvp footage:
https://youtu.be/xKCT0Cjnjws

Everything above also applied to player controlled mobs. Pets  shouldn't have problems against higher level players and players shouldn't have problems against higher level enemy pets.
1) 49 felhunter used spell lock on a 60 paladin in the previous video.
2) 48 druid used hibernate against level 60 enemy pet https://youtu.be/xldmdYS5DB4?t=185

All videos were filmed on retail vanilla and downloaded (except 36 hunter and druid ones) from warcraftmovies.com using premium mirrors since the original streams gone long time ago.

Sadly i don't know what real chance to miss should be. But current system has to go ASAP. 
It's not blizzlike.
It's not even close to vanilla experience.
It's so broken.
It's absolutely **screws** casters and pvp during leveling (battlegrounds too) is a big part of the game.

**Comment by rezzidue** [comment]
I disagree. It has always been this way. Attacking a red player to you will likely cost all your mana due to resists. I don't remember it any other way.
Yes it is biased toward melee which don't have the same harsh miss rates, but it IS World of Meleecraft (aka Roguecraft)

**Comment by Crusaderz** [comment]
> I disagree. It has always been this way. Attacking a red player to you will likely cost all your mana due to resists. I don't remember it any other way.

Then you have really bad memory. Even from battleground standpoint: blizzard made x0-x9 level range aviable for every BG bracket for a reason. Also you **totally** ignored video proofs.

> Yes it is biased toward melee which don't have the same harsh miss rates, but it IS World of Meleecraft 
Nope. Not even close.

**Comment by Crusaderz** [comment]
Another video.
46 shaman vs 60. 8 shocks used and only 1 resisted.
https://www.youtube.com/watch?v=ufizLILcCPU

So far i provided video proofs for **_all_** casters except paladins. I found video (warcraftmovies) of level 19 twinks raiding blackrock mountain and killing 60's. Paladin POV. Sometimes his hammer of justice stuns enemy so there is chance even with 41 level difference. I can post video link later if needed.

Also i found similar topic raised on nostalrius forums:
https://forum.nostalrius.org/viewtopic.php?f=45&t=3288

> Hopefully someone can confirm this **because I very specifically remember this mechanic** in Vanilla:
> 
> Your chance to hit against an enemy player (weapons & Spells) was not determined by weapon skills or difference in level. This gave a fair(er) chance to lower level characters who got jumped by higher level characters. I very specifically remember this because I knew that weapon skills were handled as "Maxed" when fighting in PvP, so I would run up to people on my mage with a staff (for that humiliating staff kill), with a weapon skill of 'low # here' and do damage proportional to what a maxed weapon skill would do. I otherwise would have been missing enemies constantly. 
> 
> I attempted to find some patch notes or some 2004-2006 articles on this but was unable to procure anything sadly. If someone knows an article that mentions this, please link it.
> 
> **This mechanic NEEDS to be recreated because even when getting jumped a by a player 3-4 levels higher than myself, my spells as a mage are being resisted at an unusually high rate**. And could a Dev confirm if they know about this mechanic?

Another comment:

> Both spells and melee are supposed to affect anyone of any level in PvP situations. It was literally one of the major selling points of WoW at the day. **I remember, plain as day, pvping on my level 22 druid with a bunch of level mid-30s and 40s specifcally because my Root was still able to effect ?? targets**. Someone even has fraps of it. 
> 
> Level 20 Mages are supposed to be able to polymorph level 60s. Level 20 Warlocks are supposed to be able to land fears. Because they're supposed to at least have a fighting chance of surviving. Again, this was one of the major selling points of wow of the day.y.
> 
> The fact that I can't hit totems with my staff is complete and utter horse shit. Weapon skill and resist are simply wrongly done. The fact that low-level melees can land attacks without fail, where low-level spellcasters can not, should be alarming enough.

> I'm going to keep looking for an article / patch note on this. I've talked with some friends playing and they remember this as well. Just looking to see if anyone else remembers / has some info on this

So yeah. I'm not only one who remember that. More interesting stuff from that forum:

> Here's a group of level 29s killing a level 60 from a mage's perspective in Vanilla:
> https://www.youtube.com/watch?v=o3GsLK2o7Xk
> 
> The mage gets a lot of resists but about ~50% of his spells land. I'm fairly sure this would not have happened on Nostalrius. Melee rolls have a mechanic where chance to miss caps at 60%, is there supposed to be something similar for spells?

Someone then linked archived blue post pointing at official chance to hit formula: https://web.archive.org/web/20060529065634/http://forums.worldofwarcraft.com/thread.aspx?fn=blizzard-archive&t=37&p=1&tmp=1#post37

But later agreed that there is something blizzard untold, since according to their formula players have 3% hit chance against someone 15 levels higher.

> Anyways, some more cursory evidence:
> https://youtu.be/jt5BepzK7oM?t=2m40s Level 39 hunter freezing trapping level 60 rogue. He would have a 1% spell hit chance if we extrapolate miss chance linearly for all level differences greater than 2.

> Indeed, your videos definitely tend to show that there was another "part" of the formula though, or at least something that wasn't explicitly told by Blizzard's post.

I raised very important topic. Invested my own time and money (WCM premium) while gathering proofs and i think i can speak on behalf of all casters who gets roflstomped right now by someone 5+ levels higher:
Devs, whether there would be something done about it?

**Comment by GuybrushGit** [comment]
The evidence is compelling, but any change to this system would be a custom formula there's nowhere near enough data to figure out the real system. What do you propose?

**Comment by Crusaderz** [comment]
@GuybrushGit

> The evidence is compelling but any change to this system would be a custom formula. There's nowhere near enough data to figure out the real system.
> What do you propose?

I totally understand the situation.

In my opinion one custom formula (current one) should be changed to another custom formula. Sure it won't be blizzlike, but it could be as close as possible to original mechanic.

For example: archived blue post https://web.archive.org/web/20060529065634/http://forums.worldofwarcraft.com/thread.aspx?fn=blizzard-archive&t=37&p=1&tmp=1#post37
stated there is **96%** chance to hit against equal level player, then -1 per level until 2 level difference. Then -7 for each level. We know that's not true.

But what if we leave 1 per level and add some coefficients based on level difference:

ROUND( 100 - ( LEVELDIFF / 10 * FLOOR( LEVELDIFF / 10 ) + LEVELDIFF + 4 ) )

![image](https://user-images.githubusercontent.com/3641625/32892342-f71ef568-cae6-11e7-81fc-bbb76ae36c25.png)

Also good for  battlegrounds. Since it's -1 chance to hit until 9 level difference.

Quote from above:

> Here's a group of level 29s killing a level 60 from a mage's perspective in Vanilla:
> https://www.youtube.com/watch?v=o3GsLK2o7Xk
> 
> The mage gets a lot of resists but about ~50% of his spells land. I'm fairly sure this would not have 

 31 level difference is a 56% chance to hit now. Sounds about right.

> 1 shock resists out of 8 shocks? 14 level difference

14  levels difference is a 81% hit chance.
 

Is there will be any harm caused by this formula? I don't think so.

Someone will escape higher level ganker by casting roots\sheep\fear on him. Working as intended.
Some really bad player is going to die against -10 levels caster. Working as intended. Also physical damage dealers already can do that.
Lower level casters ain't going to dominate world but in some cases they will be able to fight back and that was how things were back then.

**Comment by thepparker** [comment]
None of the videos you have gathered prove definitively anything is incorrect with how we currently handle level based resists. To begin with, when referring to 'max weapon skill' in PvP, this means that your CURRENT max weapon skill is used to calculate melee hit and crit chances. For example, a level 35 player has a 175 weapon skill cap. If they were a caster, and had their current weapon skill with their equipped weapon was only 20, any hit calculations against other players would be performed using their cap of 175 (not the level 60 cap of 300).

Hunter's improved concussive shot has already been proven to use physical hit calculations rather than spell hit, and that is how we implement it in our core. Therefore, spell hit is inapplicable in this case. Moreover, just because an enemy is a skull does not mean they are excessively above the current player's level. Skull is 10 or more levels, therefore it may not be as large an impact as you believe. Not all of the players in the low level gank squad you linked are necessarily level 60, not is it unfeasible for a group of low level players to kill a level 60. In fact, looking at the video, he does in fact have a large portion of resists.

In the case of the level 48 shadow priest: the shadow talent tree has talents which give 10% spell hit for shadow spells. Given he is already 48, this means his resist chance against level 60 players is already relatively small and hence he is able to land a large portion of his spells with no resists. This is similar for mages, who have multiple talents that improve their spell hit chance, particularly for arcane spells (10%, frost/fire is 6%). Warlocks have this as well, but only for affliction type spells (such as fear and corruption), which is again, 10%.

Unfortunately we can't just make up a formula when we have formulas that are well researched and established, with blue posts as well.

**Comment by Crusaderz** [comment]
@thepparker 

> None of the videos you have gathered prove definitively anything is incorrect with how we currently handle level based 

They definitively did. Try to cast something against 8 levels higher and then rewatch videos. You have no idea how it works right now on the server.

> To begin with, when referring to 'max weapon skill' in PvP, this means that your CURRENT max weapon skill is used to calculate melee hit and crit chances. For example, a level 35 player has a 175 weapon skill cap. If they were a caster, and had their current weapon skill with their equipped weapon was only 20, any hit calculations against other players would be performed using their cap of 175 (not the level 60 cap of 300).

I never refered to weapon skill at all, this is quote from nostalrius forum. I cited it with highlighted part "because I very specifically remember this mechanic". And weapon skill has nothing to do with a spell miss. Not related to the topic.

> Hunter's improved concussive shot has already been proven to use physical hit calculations rather than spell hit, and that is how we implement it in our core. Therefore, spell hit is inapplicable in this case. 

https://github.com/elysium-project/server/pull/2746#issuecomment-331998568 
look what GuybrushGit said about imp. concussive shot when i asked about resists. It's a spell and here on LH it's always reists against higher level targets due wrong formula. I have 39 hunter. Also imp. conc. shot proc resists in a videos too.

> Moreover, just because an enemy is a skull does not mean they are excessively above the current player's level. Skull is 10 or more levels, therefore it may not be as large an impact as you believe. Not all of the players in the low level gank squad you linked are necessarily level 60, not is it unfeasible for a group of low level players to kill a level 60. In fact, looking at the video, he does in fact have a large portion of resists.

Dal'rend's sword. Shows commander (rank 11) when died. What level that rogue could be? And we can see 60's in a hunters duel videos (before or after duel started) because they in the same faction.
Also out of 17 spells lvl 29 mage casted only 7 resisted against player 31 level higher. A warrior 6 levels higher resisted 8 curse of agony casts in a row when i leveled my lock last time. Nothing is wrong.

> In the case of the level 48 shadow priest: the shadow talent tree has talents which give 10% spell hit for shadow spells. Given he is already 48, this means his resist chance against level 60 players is **already relatively small**

Relatively small it was on retail vanilla, not on the mangos server. Below i explained what level 48vs60 hit chance (if you use official formula the Tseric posted back in 2006).

> Unfortunately we can't just make up a formula when we have formulas that are well researched and established, with blue posts as well

No one has ever researched and established "spell chance to hit formula". Ever. Blue post was incorrect and that's proven with videos. According to it a level 48 priest even with +10% hit should have a **34%** chance to hit against level 60. Check priest video again.


So you are going to leave broken from the beginning formula even if it ruins every casters pvp experience during leveling and low level battlegrounds? Even with all screaming evidence?

**Comment by Crusaderz** [comment]
I recorded video of my level 37 priest (+10% spellhit talent) uses rank 1 shadow word: pain vs 49 shaman. 40 casts, 11 hits.
https://youtu.be/NdZ3lDDbH-Y

Pure nonsense. Even 29 mage had better results vs 60 rogue.

+1 retail video:
38 warlock duels 52 warrior (link with timestamp in it) 16 casts, 2 misses. There is more in that video If you're willing to watch.
https://youtu.be/XqTfP-Gnj0U?t=1013

@thepparker 
Please tell, what else do you need?

**Comment by QAkvastskaft** [comment]
PVP videos cannot really be used for statistical analysis due to an extreme likelihood of cherry picking by the video maker.

https://en.wikipedia.org/wiki/Cherry_picking

**Comment by GuybrushGit** [comment]
One possibility is that level difference resist chance had a hard cap. So the formulas given by Blizzard were correct but capped at 60% or 70% chance.

**Comment by Crusaderz** [comment]
> PVP videos cannot really be used for statistical analysis due to an extreme likelihood of cherry picking by the video maker.
> 
> https://en.wikipedia.org/wiki/Cherry_picking

I would agree if there were couple casts maximum.
There is no way to lands so much spells **in a row** versus higher level players while using official formula. Cherry picking or not.

Level 38 warlock has 17% (with talents) chance to hit versus level 52. And he did land (in a video above) 10 spells in a row (icluding succubus seduction and since pets does not scale with their masters stats thats would be 7%) before he got first resisit. What chances to roll <= 17 our of 100 ten times in a row?

29 mage has 1% chance and still somehow managed to land 10 out of 17 spells vs 60 rogue.

So you have an obvious problem. Current formula is wrong and no one has ever researched how level difference affects spell hit chance in pvp. Leaving things as they are is not an option since it's ruins leveling experience \ battlegrounds for casters.

Can you guys do something?

**Comment by cnzjs** [comment]
Agreed,the PVP resistance have huge different with PVE resistance in 2004(2005).
It happened to me many times so I remember it clearly.

**Comment by ifyousayso** [comment]
In cases like this, you cannot prove anything. On retail, I soloed Zaricotl with a warlock that was 10 levels lower on the first attempt. Sure, it resisted a lot, but it was possible. Good luck doing that here.
